### PR TITLE
saga-8.632: Create IconActionBox component

### PR DESCRIPTION
**Pull request by cristian-seoane**
-Create IconActionBox component
-Add Link icon 
-Change the button children as optional

Spec: https://trovatrip.slite.com/app/channels/bbcRB2YFY8/notes/m5~J7d0Bc

Blocking: https://asanal.ink/21/549

 
https://asanal.ink/21/632

### CODE REVIEW

File: src/components/button/button.tsx [hunk 1]
```diff
@@ -68,8 +71,15 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
                     data-testid={genDataTestId('button-text-container')}
                     css={buttonText(theme, buttonType, subtext, size, icon)}
                 >
-                    <div data-testid={genDataTestId('button-text')} css={buttonChildren()}>{children}</div>
-                    {icon && <div css={iconContainer(size)}>{icon}</div> }
+                    {children && (
```
Comment (cristian-seoane) [inline src/components/button/button.tsx:74]: Just change children as optional the rest is formatting.

File: src/components/IconActionBox/IconActionBox.test.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import userEvent from '@testing-library/user-event';
+import { matchers } from '@emotion/jest';
+import IconActionBox from './IconActionBox';
+import { IconActionBoxProps } from './IconActionBox';
+import { IconName } from '../icon';
+import { ThemeProvider } from '@emotion/react';
+import theme from '../../shared/themes/theme';
+import { ComponentWidth } from '../input/componentWidth';
+
+expect.extend(matchers);
```
Comment (BrandonDenham) [inline src/components/IconActionBox/IconActionBox.test.tsx:12]: hmm, can we do this globally somewhere when test scripts are spinning up, then we don't need to do this everytime we write a test

File: src/components/IconActionBox/IconActionBox.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+interface RightAddOn {
+    icon: IconName;
+    onClick: (event: SyntheticEvent) => void;
+}
+
+export interface IconActionBoxProps extends SimpleComponent {
+    size?: ComponentWidth;
+    rightAddOn?: RightAddOn;
+    label: string;
+}
```
Comment (BrandonDenham) [inline src/components/IconActionBox/IconActionBox.tsx:27]: should move to its own types file

RightAddOn is not a good name for this, as we will quickly want a LeftAddOn that is the same interface.

File: src/components/IconActionBox/IconActionBox.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+const IconActionBox: React.FC<IconActionBoxProps> = ({
+    label,
+    size = ComponentWidth.Flexible,
+    rightAddOn,
```
Comment (BrandonDenham) [inline src/components/IconActionBox/IconActionBox.tsx:32]: maybe rightIcon for the prop name?

**Review by BrandonDenham — CHANGES_REQUESTED** [review]

File: src/components/IconActionBox/IconActionBox.test.tsx (hunk 2 above)
Comment (cristian-seoane) [reply src/components/IconActionBox/IconActionBox.test.tsx:12]: done

File: src/components/IconActionBox/IconActionBox.tsx (hunk 4 above)
Comment (cristian-seoane) [reply src/components/IconActionBox/IconActionBox.tsx:32]: done

File: src/components/IconActionBox/IconActionBox.tsx (hunk 3 above)
Comment (cristian-seoane) [reply src/components/IconActionBox/IconActionBox.tsx:27]: done

File: src/components/IconActionBox/IconActionBox.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+export interface IconActionBoxProps extends SimpleComponent {
+    size?: ComponentWidth;
+    iconButton?: IconButton;
+    label: string;
+}
+
+const IconActionBox: React.FC<IconActionBoxProps> = ({
+    label,
+    size = ComponentWidth.Flexible,
+    iconButton,
```
Comment (BrandonDenham) [inline src/components/IconActionBox/IconActionBox.tsx:32]: I think this would be better named iconButtonProps.  (same with the interface).

**Review by BrandonDenham — APPROVED** [review]
minor comment, no need for more review

File: src/components/IconActionBox/IconActionBox.tsx (hunk 5 above)
Comment (cristian-seoane) [reply src/components/IconActionBox/IconActionBox.tsx]: done!

**Pull request merged**